Source/Core/AudioCommon/AlsaSoundStream.cpp

```diff
@@ -91,134 +196,101 @@ bool AlsaSound::SetRunning(bool running)
 
 bool AlsaSound::AlsaInit()
 {
-  unsigned int sample_rate = m_mixer->GetSampleRate();
   int err;
   int dir;
-  snd_pcm_sw_params_t* swparams;
-  snd_pcm_hw_params_t* hwparams;
-  snd_pcm_uframes_t buffer_size, buffer_size_max;
+  snd_pcm_hw_params_t* hw_params;
+  const char* device;
+  snd_pcm_format_t format;
+  unsigned int channels;
+  unsigned int sample_rate;
   unsigned int periods;
 
-  err = snd_pcm_open(&handle, "default", SND_PCM_STREAM_PLAYBACK, 0);
+  m_period_size = FRAME_COUNT_MIN;
+  sample_rate = m_mixer->GetSampleRate();
```

Comment:
I don't think a lot of this changes, so it seems odd to declare it up front instead of just in line.  They can be made `const` as well:

```
const auto sample_rate = m_mixer->GetSampleRate();

m_stereo = !Config::ShouldUseDPL2Decoder();
const u32 channels = m_stereo ? 2 : 6;
const snd_pcm_format_t format = m_stereo ? SND_PCM_FORMAT_S16 : SND_PCM_FORMAT_FLOAT;
// Enable automatic format conversion for surround since soundcards don't
// usually support float format
const char* device = m_stereo ? "default" : "plug:default";
```